### PR TITLE
Re-implement the blob tx parser in a lower-level style

### PR DESCRIPTION
**Pull request by roberto-bayardo**
This implementation is more consistent with the existing parsing code: it avoids instantiating new objects and instead grabs data directly from the payload buffer as needed.

### CODE REVIEW

File: types/parse_blob_txn.go [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+	return &r
+
+}
+func (s wrapperKzgSequence) Len() int {
+	return s.num
+}
+func (s wrapperKzgSequence) At(i int) eth.KZGCommitment {
+	kzg := eth.KZGCommitment{}
+	if i >= s.num {
+		return kzg
```
Comment (0x00101010) [inline types/parse_blob_txn.go:80]: should we return a reference for the function? this way you can return nil and make it easier for caller to distinguish return?

Comment (roberto-bayardo) [reply types/parse_blob_txn.go:80]: this is part of the interface expected by the eth. library in go-kzg -- it's idiomatic in geth code at least to always pass small fixed length arrays like these by value.

File: types/txn.go [hunk 2]
```diff
@@ -80,6 +79,28 @@ func NewTxParseContext(chainID uint256.Int) *TxParseContext {
 	return ctx
 }
 
+// recover sender address after appropriately populating ctx.Sighash, ctx.R & ctx.S
+func (ctx *TxParseContext) RecoverSender(vByte byte, sender []byte) error {
+	binary.BigEndian.PutUint64(ctx.Sig[0:8], ctx.R[3])
```
Comment (0x00101010) [inline types/txn.go:84]: curious about the general encoding choices, I saw both LittleEndian & BigEndian used, is that right?

Comment (roberto-bayardo) [reply types/txn.go:84]: Unfortunately yes.. ssz uses the opposite of RLP/legacy encoding.

**Review by 0x00101010 — APPROVED** [review]

**Pull request merged**